scripts/verify-install.sh

```diff
@@ -164,13 +184,37 @@ cd $TEMP_DIR || exit 1
 # the signatures actually sign that exact manifest.
 NUM_CHECKS=0
 for signature in $SIGNATURES; do
-  echo "Verifying $signature"
-  if gpg --verify "$signature" "$MANIFEST" 2>&1 | grep -q "Good signature"; then
-    echo "Signature for $signature appears valid: "
-    gpg --verify "$signature" "$MANIFEST" 2>&1 | grep "using"
-  elif gpg --verify "$signature" 2>&1 | grep -q "No public key"; then
-    echo "Unable to verify signature $signature, no key available, skipping"
+  # Remove everything from the filename after the username.
+  USERNAME=${signature%-$VERSION.sig}
+
+  # Remove the manifest- part before the username.
+  USERNAME=${USERNAME##manifest-}
+
+  # If the user is known, they should have a key ring file with only their key.
+  KEYRING="$TEMP_DIR/$USERNAME.pgp"
+  if [[ ! -f "$KEYRING" ]]; then
+    echo "User $USERNAME does not have a known key, skipping"
     continue
+  fi
+
+  # We'll write the status of the verification to a special file that we can
+  # then inspect.
+  STATUS_FILE="$TEMP_DIR/$USERNAME.status"
+  gpg --no-default-keyring --keyring "$KEYRING" --status-fd=1 \
```

Comment:
The check if the status file exists is not here failing to detect duplicated values in the `$SIGNATURES` array leading to counting single signature multiple times.

Reply:
Correct. Though the attacker would need to upload a file with a name that leads to the same username being extracted without getting a duplicate filename in the first place. Added a check anyway.

Reply:
I assumed github itself can be an attacker (compromised or malicious) and just return broken json.